[riscv64]: FileCheck stderr: tests/codegen-llvm/ilog_known_base.rs error: CHECK: expected string not found in input

<!--
Thank you for filing a bug report! 🐛 Please provide a short summary of the bug,
along with any information you feel relevant to replicating the bug.
-->

Running `./x test tests/codegen-llvm`:

```rust
...
failures:                                                                                                             
                                                                                                                      
---- [codegen] tests/codegen-llvm/ilog_known_base.rs stdout ----                                                      
------FileCheck stdout------------------------------                                                                                                                                                                                         
                                                                                                                                                                                                                                             
------FileCheck stderr------------------------------                                                                                                                                                                                         
/home/vimer/rust/rust/tests/codegen-llvm/ilog_known_base.rs:10:12: error: CHECK: expected string not found in input                                                                                                                          
 // CHECK: %[[ICMP:.+]] = icmp ne i32 %val, 0                                                                                                                                                                                                
           ^                                                                                                                                                                                                                                 
/home/vimer/rust/rust/build/riscv64gc-unknown-linux-gnu/test/codegen-llvm/ilog_known_base/ilog_known_base.ll:7:35: note: scanning from here                                                                                                  
define { i32, i32 } @checked_ilog2(i32 noundef signext %val) unnamed_addr #0 {                                                                                                                                                               
                                  ^                                                                                                                                                                                                          
/home/vimer/rust/rust/build/riscv64gc-unknown-linux-gnu/test/codegen-llvm/ilog_known_base/ilog_known_base.ll:9:5: note: possible intended match here                                                                                         
 %.not.i = icmp eq i32 %val, 0                                                                                                                                                                                                               
    ^                                                                                                                                                                                                                                        
/home/vimer/rust/rust/tests/codegen-llvm/ilog_known_base.rs:20:12: error: CHECK: expected string not found in input                                                                                                                          
 // CHECK: %[[ICMP:.+]] = icmp ne i32 %val, 0                                                                                                                                                                                                
           ^                                                                                                                                                                                                                                 
/home/vimer/rust/rust/build/riscv64gc-unknown-linux-gnu/test/codegen-llvm/ilog_known_base/ilog_known_base.ll:26:35: note: scanning from here                                                                                                 
define { i32, i32 } @checked_ilog4(i32 noundef signext %val) unnamed_addr #0 {                                                                                                                                                               
                                  ^                                                                                                                                                                                                          
/home/vimer/rust/rust/build/riscv64gc-unknown-linux-gnu/test/codegen-llvm/ilog_known_base/ilog_known_base.ll:28:5: note: possible intended match here                                                                                        
 %.not.i = icmp eq i32 %val, 0                                                                                                                                                                                                               
    ^                                                                                                                                                                                                                                        
/home/vimer/rust/rust/tests/codegen-llvm/ilog_known_base.rs:31:12: error: CHECK: expected string not found in input                                                                                                                          
 // CHECK: %[[ICMP:.+]] = icmp ne i32 %val, 0                                                                                                                                                                                                
           ^                                                                                                          
/home/vimer/rust/rust/build/riscv64gc-unknown-linux-gnu/test/codegen-llvm/ilog_known_base/ilog_known_base.ll:46:36: note: scanning from here                                                                                                 
define { i32, i32 } @checked_ilog16(i32 noundef signext %val) unnamed_addr #0 {                                       
                                   ^                                                                                                                                                                                                         
/home/vimer/rust/rust/build/riscv64gc-unknown-linux-gnu/test/codegen-llvm/ilog_known_base/ilog_known_base.ll:48:5: note: possible intended match here                                                                                        
 %.not.i = icmp eq i32 %val, 0                                                                                        
    ^                                                                                                                 
                                                                                                                      
Input file: /home/vimer/rust/rust/build/riscv64gc-unknown-linux-gnu/test/codegen-llvm/ilog_known_base/ilog_known_base.ll                                                                                                                     
Check file: /home/vimer/rust/rust/tests/codegen-llvm/ilog_known_base.rs                                               
                                                                                                                      
-dump-input=help explains the following input dump.                                                                
Input was:                                                                                                            
<<<<<<                                                                                                                
            1: ; ModuleID = 'ilog_known_base.c7a39cb6bd3fdd60-cgu.0'                                                  
            2: source_filename = "ilog_known_base.c7a39cb6bd3fdd60-cgu.0"                                             
            3: target datalayout = "e-m:e-p:64:64-i64:64-i128:128-n32:64-S128"                                        
            4: target triple = "riscv64-unknown-linux-gnu"                                                                                                                                                                                   
            5:                                                                                                                                                                                                                               
            6: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(none) uwtable                                                                                                                               
            7: define { i32, i32 } @checked_ilog2(i32 noundef signext %val) unnamed_addr #0 {                                                                                                                                                
check:10'0                                       X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found                                                                                                                         
            8: start:                                                                                                                                                                                                                        
check:10'0     ~~~~~~~                                                                                                                                                                                                                       
            9:  %.not.i = icmp eq i32 %val, 0                                                                                                                                                                                                
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                               
check:10'1         ?                           possible intended match                                                                                                                                                                       
           10:  br i1 %.not.i, label %_RNvMs6_NtCscCqJDfSFtOR_4core3numm12checked_ilog.exit, label %bb6.i                                                                                                                                    
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                   
           11:                                                                                                                                                                                                                               
check:10'0     ~                                                                                                                                                                                                                             
           12: bb6.i: ; preds = %start                                                                                                                                                                                                       
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                      
           13:  %_37.i = tail call range(i32 0, 33) i32 @llvm.ctlz.i32(i32 %val, i1 true)                                                                                                                                                    
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                   
           14:  %_5.lhs.trunc.i = xor i32 %_37.i, 31                                                                                                                                                                                         
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                        
           15:  br label %_RNvMs6_NtCscCqJDfSFtOR_4core3numm12checked_ilog.exit                                                                                                                                                              
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                             
           16:                                                                                                                                                                                                                               
check:10'0     ~                                                                                                      
           17: _RNvMs6_NtCscCqJDfSFtOR_4core3numm12checked_ilog.exit: ; preds = %start, %bb6.i                                                                                                                                               
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                       
           18:  %_0.sroa.7.1.i = phi i32 [ undef, %start ], [ %_5.lhs.trunc.i, %bb6.i ]                                                                                                                                                      
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                     
           19:  %_0.sroa.0.1.i = phi i32 [ 0, %start ], [ 1, %bb6.i ]                                                 
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                
           20:  %0 = insertvalue { i32, i32 } poison, i32 %_0.sroa.0.1.i, 0                                           
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                 
           21:  %1 = insertvalue { i32, i32 } %0, i32 %_0.sroa.7.1.i, 1                                               
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                              
           22:  ret { i32, i32 } %1                                                                      
check:10'0     ~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                      01:43:30 [287/1861]
           23: }                                                                                                      
check:10'0     ~~                                                                                                     
           24:                                                                                                        
check:10'0     ~                                                                                                      
           25: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(none) uwtable        
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                              
           26: define { i32, i32 } @checked_ilog4(i32 noundef signext %val) unnamed_addr #0 {                                                                                                                                                
check:10'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                            
check:20'0                                       X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found                                                                                                                         
           27: start:                                                                                                                                                                                                                        
check:20'0     ~~~~~~~                                                                                                                                                                                                                       
           28:  %.not.i = icmp eq i32 %val, 0                                                                                                                                                                                                
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                               
check:20'1         ?                           possible intended match                                                                                                                                                                       
           29:  br i1 %.not.i, label %_RNvMs6_NtCscCqJDfSFtOR_4core3numm12checked_ilog.exit, label %bb6.i                                                                                                                                    
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                   
           30:                                                                                                                                                                                                                               
check:20'0     ~                                                                                                                                                                                                                             
           31: bb6.i: ; preds = %start                                                                                                                                                                                                       
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                      
           32:  %_37.i = tail call range(i32 0, 33) i32 @llvm.ctlz.i32(i32 %val, i1 true)                                                                                                                                                    
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                   
           33:  %_5.lhs.trunc.i = lshr i32 %_37.i, 1                                                                                                                                                                                         
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                        
           34:  %_59.i1 = xor i32 %_5.lhs.trunc.i, 15                                                                                                                                                                                        
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                       
           35:  br label %_RNvMs6_NtCscCqJDfSFtOR_4core3numm12checked_ilog.exit                                                                                                                                                              
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                             
           36:                                                                                                        
check:20'0     ~                                                                                                                                                                                                                             
           37: _RNvMs6_NtCscCqJDfSFtOR_4core3numm12checked_ilog.exit: ; preds = %start, %bb6.i                        
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                              
           38:  %_0.sroa.7.1.i = phi i32 [ undef, %start ], [ %_59.i1, %bb6.i ]                                                                                                                                                              
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                      
           39:  %_0.sroa.0.1.i = phi i32 [ 0, %start ], [ 1, %bb6.i ]                                                 
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                
           40:  %0 = insertvalue { i32, i32 } poison, i32 %_0.sroa.0.1.i, 0                                                                                                                                                                  
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                          
           41:  %1 = insertvalue { i32, i32 } %0, i32 %_0.sroa.7.1.i, 1                                               
check:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                     
           42:  ret { i32, i32 } %1                                                           
...
check:31'0     ~                                                                                                      
           65: ; Function Attrs: mustprogress nocallback nofree nosync nounwind speculatable willreturn memory(none)  
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
           66: declare i32 @llvm.ctlz.i32(i32, i1 immarg) #1                                                          
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                         
           67:                                                                                                                                                                                                                               
check:31'0     ~                                                                                                                                                                                                                             
           68: attributes #0 = { mustprogress nofree norecurse nosync nounwind willreturn memory(none) uwtable "target-cpu"="generic-rv64" "target-features"="+m,+a,+zaamo,+zalrsc,+f,+zicsr,+d,+f,+zicsr,+c,+zca,+zicsr,+zifencei" }        
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~       
           69: attributes #1 = { mustprogress nocallback nofree nosync nounwind speculatable willreturn memory(none) }                                                                                                                       
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                      
           70:                                                                                                                                                                                                                               
check:31'0     ~                                                                                                                                                                                                                             
           71: !llvm.module.flags = !{!0, !1, !2, !3}                                                                                                                                                                                        
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                       
           72: !llvm.ident = !{!4}                                                                                                                                                                                                           
check:31'0     ~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                          
           73:                                                                                                                                                                                                                               
check:31'0     ~                                                                                                                                                                                                                             
           74: !0 = !{i32 8, !"PIC Level", i32 2}                                                                                                                                                                                            
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                           
           75: !1 = !{i32 1, !"Code Model", i32 3}                                                                                                                                                                                           
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                          
           76: !2 = !{i32 7, !"uwtable", i32 2}                                                                                                                                                                                              
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                             
           77: !3 = !{i32 1, !"target-abi", !"lp64d"}                                                                                                                                                                                        
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                       
           78: !4 = !{!"rustc version 1.98.0-dev"}                                                                                                                                                                                           
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                   
>>>>>>                                                                                                                                                                                                                                       
------------------------------------------                                                                            
                                                                                                                                                                                                                                             
---- [codegen] tests/codegen-llvm/ilog_known_base.rs stdout end ----                                                                                                                                                                         
                                                                                                                      
failures:                                                                                                             
    [codegen] tests/codegen-llvm/ilog_known_base.rs                                                                   
                                                                                                                                                                                                                                             
test result: FAILED. 843 passed; 1 failed; 231 ignored; 0 measured; 0 filtered out; finished in 182.68s               
                                                                                                                      
Some tests failed in compiletest suite=codegen-llvm mode=codegen host=riscv64gc-unknown-linux-gnu target=riscv64gc-unknown-linux-gnu       

```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
rustc version 1.98.0-dev
```

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->

Detected by [RuyiCI](https://github.com/ruyici/rustci4rv) and opening this to track this.

--- a/tests/codegen-llvm/ilog_known_base.rs
+++ b/tests/codegen-llvm/ilog_known_base.rs
@@ -7,7 +7,7 @@
 // CHECK-LABEL: @checked_ilog2
 #[no_mangle]
 pub fn checked_ilog2(val: u32) -> Option<u32> {
-    // CHECK: %[[ICMP:.+]] = icmp ne i32 %val, 0
+    // CHECK: icmp {{ne|eq}} i32 %val, 0
     // CHECK: %[[CTZ:.+]] = tail call range(i32 0, 33) i32 @llvm.ctlz.i32(i32 %val, i1 true)
     // CHECK: %[[LOG2:.+]] = xor i32 %[[CTZ]], 31
     val.checked_ilog(2)
@@ -17,7 +17,7 @@ pub fn checked_ilog2(val: u32) -> Option<u32> {
 // CHECK-LABEL: @checked_ilog4
 #[no_mangle]
 pub fn checked_ilog4(val: u32) -> Option<u32> {
-    // CHECK: %[[ICMP:.+]] = icmp ne i32 %val, 0
+    // CHECK: icmp {{ne|eq}} i32 %val, 0
     // CHECK: %[[CTZ:.+]] = tail call range(i32 0, 33) i32 @llvm.ctlz.i32(i32 %val, i1 true)
     // CHECK: %[[DIV2:.+]] = lshr i32 %[[CTZ]], 1
     // CHECK: %[[LOG4:.+]] = xor i32 %[[DIV2]], 15
@@ -28,9 +28,9 @@ pub fn checked_ilog4(val: u32) -> Option<u32> {
 // CHECK-LABEL: @checked_ilog16
 #[no_mangle]
 pub fn checked_ilog16(val: u32) -> Option<u32> {
-    // CHECK: %[[ICMP:.+]] = icmp ne i32 %val, 0
+    // CHECK: icmp {{ne|eq}} i32 %val, 0
     // CHECK: %[[CTZ:.+]] = tail call range(i32 0, 33) i32 @llvm.ctlz.i32(i32 %val, i1 true)
     // CHECK: %[[DIV4:.+]] = lshr i32 %[[CTZ]], 2
-    // CHECK: %[[LOG16:.+]] = xor i32 %[[DIV2]], 7
+    // CHECK: %[[LOG16:.+]] = xor i32 %[[DIV4]], 7
     val.checked_ilog(16)
 }

